core/storage/blog_statistics/gae_models.py

```diff
@@ -0,0 +1,999 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing statsistics data related to blog posts."""
+
+from __future__ import annotations
+
+import datetime
+
+from core import utils
+from core.platform import models
+
+from typing import Dict, List
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+
+(base_models, ) = models.Registry.import_models([models.Names.BASE_MODEL])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class BlogPostViewedEventLogEntryModel(base_models.BaseModel):
+    """An event triggered when a blog post is viewed by any user."""
+
+    # ID of blog post currently being viewed.
+    blog_post_id = datastore_services.StringProperty(
+        indexed=True, required=True)
+    # User ID of the author of the blog post.
```

Comment:
If you store the blog post ID then why do you need this field for the raw event log? I can imagine it in a summary stat but is it superfluous here?

Reply:
Added

Reply:
Ah ok. I think this deserves a code comment to explain that "Although we can get this from the blog post ID, we include it here because ..."

That said, shouldn't this logically be handled instead by the Beam job loading all the blog models and event models at the start, and then matching event models to blog models as needed? /cc @vojtechjelinek for confirmation -- I think this is what we have done previously.

Reply:
Yes exactly, you can get models using `GetModels` then generate pair of id and the model and match the pairs by using `CoGroupByKey`